src/main/java/com/google/sps/servlets/LogoutServlet.java

```diff
@@ -21,16 +21,55 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
+import javax.servlet.http.Cookie;
+import com.google.appengine.api.utils.SystemProperty;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
 /** Servlet that invalidates the user's session after they log out. */
 @WebServlet("/logout")
 public class LogoutServlet extends HttpServlet {
 
     @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //log the user out if we're on the development server
+        if (SystemProperty.environment.value() != SystemProperty.Environment.Value.Production) {
+            UserService userService = UserServiceFactory.getUserService();
+            String referrer = request.getHeader("referer");
+            response.sendRedirect(userService.createLogoutURL(referrer));
+
+            if(request.getSession(false) != null) {
+                request.getSession(false).invalidate();
+            }
+
+            return;
+        }
+
+        Cookie[] cookies = request.getCookies();
+
+        //user has not logged in, no cookies have been created
+        if(cookies == null) {
+            return;
+        }
+        
+        for(Cookie cookie : cookies) {
+            //Instead of the user clicking the logout link generated by the Authentication API
+            // we can just delete the cookies associated with the log in so that the user isn't logged out of
+            // Google completely.
+            // Reference: http://ptspts.blogspot.com/2011/12/how-to-log-out-from-appengine-app-only.html
+            if(cookie.getName().equals("dev_appserver_login") || cookie.getName().equals("SACSID")) {
+                cookie.setValue("");
```

Comment:
I might be wrong, but today during the podsync when you said that js interprets an empty string as the homepage, does that apply to this cookie being set to an empty string?

Reply:
I don't think so. I think that only applied to links.